src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -280,15 +295,29 @@ protected void checkAnswers(Element el, IBaseDataObject payload, @Nullable List<
 
 
         // Check each extract
-        String extractCountStr = el.getChildTextTrim("extractCount");
+        int extractCount = JDOMUtil.getChildIntValue(el, "extractCount");
+        long numExtractElements =
+                el.getChildren().stream().filter(c -> c.getName().startsWith("extract") && !c.getName().startsWith("extractCount")).count();
+        if (extractCount > -1 || numExtractElements > 0) {
+            if (!skipExtractCountValidation) {
+                assertEquals(Math.max(extractCount, 0), Math.toIntExact(numExtractElements),
+                        "Expected extractCount in " + tname + " not equal to number of <extract#> elements");
+            } else {
+                if (Math.toIntExact(numExtractElements) != extractCount) {
+                    logger.warn("Expected extractCount and actual <extract#> count in {} not equal. Expected: {} Actual: {}", tname,
+                            Math.max(extractCount, 0), numExtractElements);
+                }
+            }
+        }
 
         if (payload.hasExtractedRecords()) {
             List<IBaseDataObject> extractedChildren = payload.getExtractedRecords();
             int foundCount = extractedChildren.size();
 
-            if (extractCountStr != null) {
-                assertEquals(Integer.parseInt(extractCountStr), foundCount,
-                        String.format("Number of extracted children in '%s' is %s, not expected %s", tname, foundCount, extractCountStr));
+            if (extractCount > -1) {
+                assertEquals(extractCount, foundCount,
```

Comment:
Should this get the same `skipExtractCountValidation` treatment as above? (warn vs assert)

Reply:
I could, though that is kind of out of the scope of this PR. The check above this verifies answer files have matching counts, I'm not sure if we want to have the option to "skip validation" when actually checking the answerFile vs the explicit payload extracted records count

Reply:
Ah - you're 100% right.  I lost track of which count the various variables were referencing.